### PR TITLE
add functional tests for windows

### PR DESCRIPTION
**Pull request by sowu880**
Main changes:
1. Add functional tests for vscode client in OCIArtifactFunctionaltests.cs. Simulate different situations and listed in the inline comments at the top of the cs file.
2. Update functional pipeline, refine registry building stage in windows vm.

Useful Concept:
    Sequence number: 
       the sequence of each layer starts from 1. (e.g. sequence of base layer is 1, user layer is 2). If there is only one layer for an image, the sequence of the layer should be 1 or -1. (-1 mains the layer is not packed by our tool). If the image has multi layers, the sequence should be continuous and non-repeatable. (note! : Only layer with sequence 1 can be the base layer. )

### CODE REVIEW

File: release.yml [hunk 1]
```diff
@@ -127,7 +127,10 @@ stages:
         artifactName: FhirConverterBuild
         downloadPath: $(System.DefaultWorkingDirectory)
     - script: |
-        docker run --rm -d -p 5000:5000 --name registry stefanscherer/registry-windows:2.6.2
+        git clone -q https://github.com/sowu880/dockerfiles-windows.git
+        cd dockerfiles-windows/registry/
+        docker build -t registry-windows:2.7.1 .
+        docker run --rm -d -p 5000:5000 --name registry registry-windows:2.7.1
```
Comment (ACMoretxj) [inline release.yml:133]: Just for confirmation, if we run a docker in daemon mode, will the azure pipeline wait for the container to finish running?

File: src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/OrasUtility.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                StartInfo = new ProcessStartInfo(orasFileName),
+            };
+
+            process.StartInfo.Arguments = command;
+            process.StartInfo.RedirectStandardError = true;
+            process.EnableRaisingEvents = true;
+            process.Start();
+
+            StreamReader errStreamReader = process.StandardError;
+            process.WaitForExit(30000);
```
Comment (ACMoretxj) [inline src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/OrasUtility.cs:26]: The constant value `30000` can be extracted out as a static readonly class field for unified management.

Comment (sowu880) [reply src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/OrasUtility.cs:26]: Constants is an internal class could not be used here. Maybe hard code here is fine in test project?

Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/OrasUtility.cs:26]: I see. I believe the hard coding for tests is ok here.

File: release.yml (hunk 1 above)
Comment (ACMoretxj) [reply release.yml:133]: By teams discussion there is no issue for the docker-in-pipeline :)

**Review by ACMoretxj — APPROVED** [review]

File: release.yml [hunk 3]
```diff
@@ -127,7 +127,10 @@ stages:
         artifactName: FhirConverterBuild
         downloadPath: $(System.DefaultWorkingDirectory)
     - script: |
-        docker run --rm -d -p 5000:5000 --name registry stefanscherer/registry-windows:2.6.2
+        git clone -q https://github.com/sowu880/dockerfiles-windows.git
```
Comment (moria97) [inline release.yml:130]: nit: can we rename the repository to some meaningful name?

**Review by moria97 — APPROVED** [review]

**Pull request merged**